### ISSUE TITLE
Nvidia Optimus option when choosing drivers

### ISSUE THREAD

**Issue by anAlpiner** [archlinux/archinstall#2580]
# Copy pasted from discord
my suggestion is:
add optimus option after choosing desktop environment (in desktop profile) 
currently we have:
----
all open source 
nvidia
nvidia optimus (For integrated and discrete graphics, like Nvidia + Intel)
intel
amd ATI 
vmware/vbox
...
----
So my suggestion is adding nvidia optimus to that list
----
all open source 
nvidia
nvidia optimus (For integrated and discrete graphics, like Nvidia + Intel)
intel
amd ATI 
vmware/vbox
...
----
(I have nvidia + intel graphics machine so thats why i need that, using only 1 graphics is not my way, because i want to use nvidia only for games, not all time)

**Comment by Torxed** [comment]
I feel this is something that could be configured post installation, or are there steps that's needed during installation?

**Comment by Tarballwalf** [comment]
I think the best approach would be to allow users to select 2 profiles if needed. This would mostly just free up post installation stuff.

**Comment by EK-92** [comment]
👋 howdy

### TLDR: just use `all open-source` since Nvidia is now available to all
___

IMO setting up dual GPUs should be up to user post-install due to sheer complexity and possibilities.

Possible GPU combinations include but aren't limited to:
| built-in | external |
|---|---|
| AMD | AMD |
| AMD | Nvidia |
| Intel | AMD |
| Intel | Nvidia |

That's just what we had so far for x86/x64, eventually we might have external Intel GPUs for laptops, as well as whatever happens with ARM devices like latest macbooks.

Behavior-wise you can:
- disable external GPU to reduce power consumption
- have it as hybrid in case you want a more flexible on-demand processing power (which I believe is default in most cases)
- disable internal GPU and have external one handle everything (which is what I did for my AMD+Nvidia laptop)

That's 3 possible ways to handle it for 4 possible combinations of GPUs, which ones do we include? My answer is none since the point of archinstall is to make a simple un-bloated Arch installation, but maybe others disagree. And I'm not even listing possible GPU switching methods since they differ based on GPU combinations (see arch wiki links below).

### Main points:
- Yes, you do need drivers for both Intel/Nvidia.
- Yes, archinstall probably should allow for selecting multiple drivers arbitrarily.
- No, you don't need some specific drivers for your hybrid setup, but you _might_ need a patched display manager for it, such as [gdm-prime (AUR)](https://aur.archlinux.org/packages/gdm-prime) for Gnome. The gdm-prime makes use of Canonical's "launch with external GPU" patch which vanilla GDM doesn't do.

___

Some relevant arch wiki pages:
[Hybrid graphics](https://wiki.archlinux.org/title/Hybrid_graphics)
[Nvidia Optimus](https://wiki.archlinux.org/title/NVIDIA_Optimus)
[Prime](https://wiki.archlinux.org/title/PRIME)

**Comment by Torxed** [comment]
We'll close this, as I agree. The problem is complicated enough that it's better left to the user post-install. At least for the foreseeable future.
